src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -0,0 +1,469 @@
+DEFAULT_TOKEN_TYPE = 'PAT'
+
+DEFAULT_TIMEOUT_IN_SEC = 60 * 60  # 60 minutes
+DEFAULT_CPU = 2
+ALLOWED_TASK_FILE_TYPES = ('.yaml', '.toml', '.json', '.sh', '.bash', '.zsh', '.ps1', '.ps',
+                           '.cmd', '.bat', '.ts', '.js', '.php', '.py', '.rb', '.lua')
+
+
+def acr_task_create(cmd,  # pylint: disable=too-many-locals
+                    client,
+                    task_name,
+                    registry_name,
+                    context_path,
+                    file,
+                    git_access_token=None,
+                    image_names=None,
+                    status='Enabled',
+                    os_type=OS.linux.value,
+                    cpu=DEFAULT_CPU,
+                    timeout=DEFAULT_TIMEOUT_IN_SEC,
+                    values=None,
+                    source_trigger_name=None,
+                    commit_trigger_enabled=True,
+                    branch='master',
+                    no_push=False,
+                    no_cache=False,
+                    arg=None,
+                    secret_arg=None,
+                    set_value=None,
+                    set_secret=None,
+                    base_image_trigger_name=None,
+                    base_image_trigger_enabled=True,
+                    base_image_trigger_type='Runtime',
+                    resource_group_name=None):
+    if commit_trigger_enabled and not git_access_token:
+        raise CLIError("Commit trigger needs to be disabled [--commit-trigger-enabled False] "
+                       "if no --git-access-token is provided.")
+
+    if file.endswith(ALLOWED_TASK_FILE_TYPES):
+        step = FileTaskStep(
+            task_file_path=file,
+            values_file_path=values,
+            context_path=context_path,
+            values=(set_value if set_value else []) + (set_secret if set_secret else [])
+        )
+    else:
+        step = DockerBuildStep(
+            image_names=image_names,
+            is_push_enabled=not no_push,
+            no_cache=no_cache,
+            docker_file_path=file,
+            arguments=(arg if arg else []) + (secret_arg if secret_arg else []),
+            context_path=context_path
+        )
+
+    registry, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    source_control_type = SourceControlType.visual_studio_team_service.value
+    if context_path is not None and 'GITHUB.COM' in context_path.upper():
+        source_control_type = SourceControlType.github.value
+
+    task_create_parameters = Task(
+        location=registry.location,
+        step=step,
+        platform=PlatformProperties(
+            os=os_type,
+            architecture=Architecture.amd64.value
+        ),
+        status=status,
+        timeout=timeout,
+        agent_configuration=AgentProperties(
+            cpu=cpu
+        ),
+        trigger=TriggerProperties(
+            source_triggers=[
```

Comment:
If `source_triggers` is an array, how do we support adding multiple of them?

Reply:
We don't currently.